### ISSUE TITLE
[FR] Inspection for "call to a member function on a non-object"

### ISSUE THREAD

**Issue by orklah** [kalessil/phpinspectionsea#1249]
Description :
--
We already have an inspection that will check for null in variables and warn about Null Pointer Exception when we call a method on this variable.

I would like to suggest the same kind of inspection but for "Call to a member function on an non-object"

It happens every time there is a scalar in a variable and we make a method call on it.

Example
--
```php
class a{
    public function test(){

    }
}

/**
 * @param int|a $a
 */
function foo6($a){
    $a->test();
}
```
This should throw a warning for probable call on a non-object, but typically, every case matched with the NPE inspection should be matched too, but with scalar instead of null

Note: to help using this inspection, it would be cool to follow the same pattern as this suggestion:
https://github.com/kalessil/phpinspectionsea/issues/1203

It suggests that there should be a settings for those cases:
 - Trigger when all of the types provoke a call on non-object (as opposed to the normal behaviour: Trigger when at least one type provoke a call on non-object )
 - Trigger when one of the types is "mixed"

**Comment by kalessil** [comment]
Currently, this is too complex, perhaps in future versions.
